src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -154,11 +160,12 @@ def move_to_slice(self, target_slice_idx, target_slice_length):
                 candidates.sort(key=lambda c: abs(target_slice_idx - c))
                 self.program_counter = candidates[0]
 
-    def record_trace(self, target_slice_length, slice_idx=None):
+    def record_trace(self, target_slice_length, slice_idx=None, slice_type=None):
```

Comment:
Adds a new, optional parameter that lets callers override the slice type for a trace entry.